build-tools/packages/build-cli/src/commands/merge/branches.ts

```diff
@@ -107,14 +99,18 @@ export default class MergeBranch extends BaseCommand<typeof MergeBranch> {
 			// TODO: notify the author
 		}
 
-		const lastMergedCommit = await this.gitRepo.getMergeBase(flags.source, flags.target);
+		const lastMergedCommit = await this.gitRepo.getMergeBaseRemote(
+			flags.source,
+			this.remote,
+			`refs/remotes/${this.remote}/${flags.target}`,
```

Comment:
I find the logic a lot easier to follow if you use local branches wherever possible. You don't have to change this necessarily, but if you're going to use strings like refs/remotes/${this.remote}/${flags.target}, then I suggest making them variables with names like remoteTargetBranch. That makes it more obvious what the string represents.